src/index.ts

```diff
@@ -22,6 +22,46 @@ function printVersion(): void {
   }
 }
 
+const GLOBAL_BOOLEAN_FLAGS = {
+  'no-input': {
+    coerce: envToBool,
+    default: isCI,
+    describe: 'Suppresses all user prompts',
+    global: true,
+  },
+  'dry-run': {
+    coerce: envToBool,
+    // TODO(byk): Deprecate this in favor of CRAFT_DRY_RUN
+    default: process.env.DRY_RUN,
+    global: true,
+    describe: 'Dry run mode: do not perform any real actions',
+  },
+};
+
+/**
+ * This function is to pre-process and fix one of yargs' shortcomings:
+ * We want to use some flags as booleans: just their existence on the CLI should
+ * set them to true. That said since we also allow setting them through
+ * environment variables, we need to parse many string values that would set
+ * them to false. Moreover, we want to be able to override already-set env
+ * variables with the `--flag=no` kind of notation (using the `=` symbol) but
+ * not via the positional argument notation (`--flag no`). The only way to do
```

Comment:
Feels really special-casey. :-/

Reply:
Yeah, it is. Right now it is not easy to replace `yargs` with anything else but I'll be on the look.